src/main/java/com/iexec/sms/tee/EnableIfTeeProviderDefinition.java

```diff
@@ -0,0 +1,11 @@
+package com.iexec.sms.tee;
+
+import com.iexec.common.tee.TeeEnclaveProvider;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.RUNTIME)
+public @interface EnableIfTeeProviderDefinition {
+    TeeEnclaveProvider[] providers();
```

Comment:
Why is it an array here :) ?

Reply:
This way, we could ensure the following points:
- if the SMS ever evolves to be able to boot without any TEE provider, then we could easily tag beans to be added only for `Gramine` or `Scone` via `@EnableIfTeeProviderDefinition(providers = {GRAMINE, SCONE})`;
- if we ever support other providers, then some beans could be loaded only for some providers, not all.

These cases are hypothetical, but that's not a big deal to support them.